### PR TITLE
[DPMMA-1861] Duplicate segment clone action

### PR DESCRIPTION
**Pull request by tomekkowalczyk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [yes ]
| New feature/enhancement? (use the a.x branch)      | [x ]
| Deprecations?                          | [ x]
| BC breaks? (use the c.x branch)        | [x ]
| Automated tests included?              | [ yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
When re-saving the segment, duplicates are created, this is because the action attribute in the form is not changed after saving the cloned form

image

#### Steps to test this PR:

1. Clone Segment https://mautic.ddev.site/index_dev.php/s/segments/clone/{id_segment}
2. save Changes
3. save changes again

### CODE REVIEW

**Review by kuzmany — DISMISSED** [review]
Face the same issue and it looks like this fixes it 👍

**Comment by npracht** [comment]
@galvani can you please review code and test it ? Thanks :)`

File: app/bundles/LeadBundle/Controller/ListController.php [hunk 1]
```diff
@@ -347,6 +347,8 @@ private function createSegmentModifyResponse(LeadList $segment, array $postActio
                             'objectId'     => $segment->getId(),
                         ]);
 
+                        $form = $segmentModel->createForm($segment, $this->get('form.factory'), $postActionVars['returnUrl']);
+
```
Comment (galvani) [inline app/bundles/LeadBundle/Controller/ListController.php:351]: You need to fix the php doc signature of Mautic\LeadBundle\Model\ListModel::createForm()
. to make phpstan happy at least:
```php
    /**
     * {@inheritdoc}
     *
     * @param       $entity
     * @param       $formFactory
     * @param array $options
     *
     * @return mixed
     *
     * @throws \Symfony\Component\HttpKernel\Exception\NotFoundHttpException
     */
    public function createForm($entity, $formFactory, ?string $action = null, $options = [])
    {
```

**Review by galvani — CHANGES_REQUESTED** [review]
just a minor chanhe for phpstan ;-) thanks for the solution/

**Comment by sdoering** [comment]
Was not able to test this as I ran into an error twice: 

image

_Not sure about potential reasons for this_.

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well

I tested in the local instance because I had an error in gitpod.

**Comment by tomekkowalczyk** [comment]
@sdoering I correct PHPStan ;)

**Review by escopecz — APPROVED** [review]
Thanks! 👍

**Pull request merged**